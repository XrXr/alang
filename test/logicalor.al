main :: proc () {
    if true || true {
        puts("basic 1\n")
    }
    if false || true {
        puts("basic 2\n")
    }
    if false || false {
        tripwire()
    } else {
        puts("basic 3\n")
    }
    if true || tripwire() || tripwire() || tripwire() {
        puts("shortcircuit 4\n")
    }
    e := tick1() || tick2() || tick3() || tripwire()
    if e == true {
        puts("shortcircuit 5\n")
    }
}

tripwire :: proc () -> bool {
    puts("bad\n")
    return false
}

tick1 :: proc () -> bool {
    puts("tick 1\n")
    return false
}

tick2 :: proc () -> bool {
    puts("tick 2\n")
    return false
}

tick3 :: proc () -> bool {
    puts("tick 3, return true\n")
    return true
}
