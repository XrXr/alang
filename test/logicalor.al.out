basic 1
basic 2
basic 3
shortcircuit 4
tick 1
tick 2
tick 3, return true
shortcircuit 5
